contracts/oracles/UniswapV2Oracle.sol

```diff
@@ -0,0 +1,276 @@
+        return _pairs.values();
+    }
+
+    /**
+     * @notice The function to get the number of observations of a pair
+     * @param pair_ the pair address
+     * @return the number of oracle observations
+     */
+    function getPairRounds(address pair_) public view returns (uint256) {
+        return _pairInfos[pair_].blockTimestamps.length;
+    }
+
+    /**
+     * @notice The function to get the exact observation of a pair
+     * @param pair_ the pair address
+     * @param round_ the observation index
+     * @return the prices0Cumulative of the observation
+     * @return the prices1Cumulative of the observation
+     * @return the timestamp of the observation
+     */
+    function getPairInfo(
+        address pair_,
+        uint256 round_
+    ) public view returns (uint256, uint256, uint256) {
+        PairInfo storage _pairInfo = _pairInfos[pair_];
+
+        return (
+            _pairInfo.prices0Cumulative[round_],
+            _pairInfo.prices1Cumulative[round_],
+            _pairInfos[pair_].blockTimestamps[round_]
+        );
+    }
+
+    /**
+     * @notice The function to set the time window of TWAP
+     * @param newTimeWindow_ the new time window value in seconds
+     */
+    function _setTimeWindow(uint256 newTimeWindow_) internal {
+        require(newTimeWindow_ > 0, "UniswapV2Oracle: time window can't be 0");
+
+        timeWindow = newTimeWindow_;
+    }
+
+    /**
+     * @notice The function to add multiple tokens paths for the oracle to observe. Every token may only have a single path
+     * @param paths_ the array of token paths to add
+     */
+    function _addPaths(address[][] memory paths_) internal {
+        uint256 numberOfPaths_ = paths_.length;
+
+        for (uint256 i = 0; i < numberOfPaths_; i++) {
+            uint256 pathLength_ = paths_[i].length;
+
+            require(pathLength_ >= 2, "UniswapV2Oracle: path must be longer than 2");
+
+            address tokenIn_ = paths_[i][0];
+
+            require(_paths[tokenIn_].length == 0, "UniswapV2Oracle: path already registered");
+
+            for (uint256 j = 0; j < pathLength_ - 1; j++) {
+                (bool exists_, address pair_) = _pairExists(paths_[i][j], paths_[i][j + 1]);
+                require(exists_, "UniswapV2Oracle: uniswap pair doesn't exist");
+
+                _pairs.add(pair_);
+                _pairInfos[pair_].refs++;
+            }
+
+            _paths[tokenIn_] = paths_[i];
+        }
+
+        updatePrices();
+    }
+
+    /**
+     * @notice The function to remove multiple token paths from the oracle. Unregisters the pairs as well
+     * @param tokenIns_ The array of token addresses to remove
+     */
+    function _removePaths(address[] memory tokenIns_) internal {
+        uint256 numberOfPaths_ = tokenIns_.length;
+
+        for (uint256 i = 0; i < numberOfPaths_; i++) {
+            uint256 pathLength_ = _paths[tokenIns_[i]].length;
+
+            if (pathLength_ == 0) {
+                return;
```

Comment:
I think it should be `continue` instead of `return`

Reply:
Yes